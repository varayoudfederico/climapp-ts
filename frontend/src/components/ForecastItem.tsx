import WeatherIcon from "./WeatherIcon";
import { getFechaString, getTemperaturaString } from "../utils/Utils";
import { Dia } from "../types/types";

interface ForecastItemProps {
	dia: Dia;
}

const ForecastItem = ({ dia }: ForecastItemProps) => {
	const { icon, dt, description, max, min } = dia;
	return (
		<div>
			<div
				className={"card"}
				style={{
					display: "flex",
					justifyContent: "space-between",
					padding: "12px",
				}}
			>
				<div
					style={{
						display: "flex",
						justifyContent: "space-between",
					}}
				>
					<div
						style={{
							display: "flex",
							justifyContent: "column",
							marginRight: "12px",
							alignItems: "center",
						}}
					>
						<WeatherIcon iconText={icon} size="16px" color="#7754F8" />
					</div>
					<div
						style={{
							display: "flex",
							justifyContent: "column",
						}}
					>
						<div>
							<div style={{ textTransform: "capitalize" }}>
								{getFechaString(dt)}
							</div>
							<div style={{ color: "#aaaaaa", textTransform: "capitalize" }}>
								{description}
							</div>
						</div>
					</div>
				</div>

				<div
					style={{
						display: "flex",
						alignItems: "center",
					}}
				>
					<div>
						<div style={{ color: "#000" }}>
							{getTemperaturaString(max)}{" "}
							<span style={{ color: "#aaaaaa", marginLeft: "8px" }}>
								{getTemperaturaString(min)}
							</span>
						</div>
					</div>
				</div>
			</div>
		</div>
	);
};

export default ForecastItem;
