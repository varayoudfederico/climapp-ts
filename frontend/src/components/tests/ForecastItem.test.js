import "@testing-library/jest-dom";
import { render, screen, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import ForecastItem from "../ForecastItem";

afterEach(() => {
	cleanup();
});

test("muestra un ForecastItem correctamente", () => {
	const mockForecast = {
		description: "muy nuboso",
		dt: 1617552000,
		icon: "04d",
		max: 26.87,
		min: 16.06,
	};
	render(<ForecastItem dia={mockForecast}></ForecastItem>);
	expect(screen.getByText("domingo 04")).toBeInTheDocument();
	expect(screen.getByText("muy nuboso")).toBeInTheDocument();
	expect(screen.getByText("27°")).toBeInTheDocument();
	expect(screen.getByText("16°")).toBeInTheDocument();
});

test("coincide con el snapshot de ForecastItem", () => {
	const mockData = {
		description: "muy nuboso",
		dt: 1617552000,
		icon: "04d",
		max: 26.87,
		min: 16.06,
	};
	const tree = renderer
		.create(<ForecastItem dia={mockData}></ForecastItem>)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
