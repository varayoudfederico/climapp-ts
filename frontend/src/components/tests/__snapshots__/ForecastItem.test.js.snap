// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`coincide con el snapshot 1`] = `
<div>
  <div
    className="card"
    style={
      Object {
        "display": "flex",
        "justifyContent": "space-between",
        "padding": "12px",
      }
    }
  >
    <div
      style={
        Object {
          "display": "flex",
          "justifyContent": "space-between",
        }
      }
    >
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "justifyContent": "column",
            "marginRight": "12px",
          }
        }
      >
        <div
          style={
            Object {
              "color": "#7754F8",
              "fontSize": "16px",
            }
          }
        >
          <i
            className="wi wi-cloudy wi-fw"
          />
        </div>
      </div>
      <div
        style={
          Object {
            "display": "flex",
            "justifyContent": "column",
          }
        }
      >
        <div>
          <div
            style={
              Object {
                "textTransform": "capitalize",
              }
            }
          >
            domingo 04
          </div>
          <div
            style={
              Object {
                "color": "#aaaaaa",
                "textTransform": "capitalize",
              }
            }
          >
            muy nuboso
          </div>
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
        }
      }
    >
      <div>
        <div
          style={
            Object {
              "color": "#000",
            }
          }
        >
          27째
           
          <span
            style={
              Object {
                "color": "#aaaaaa",
                "marginLeft": "8px",
              }
            }
          >
            16째
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`coincide con el snapshot de ForecastItem 1`] = `
<div>
  <div
    className="card"
    style={
      Object {
        "display": "flex",
        "justifyContent": "space-between",
        "padding": "12px",
      }
    }
  >
    <div
      style={
        Object {
          "display": "flex",
          "justifyContent": "space-between",
        }
      }
    >
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "justifyContent": "column",
            "marginRight": "12px",
          }
        }
      >
        <div
          style={
            Object {
              "color": "#7754F8",
              "fontSize": "16px",
            }
          }
        >
          <i
            className="wi wi-cloudy wi-fw"
          />
        </div>
      </div>
      <div
        style={
          Object {
            "display": "flex",
            "justifyContent": "column",
          }
        }
      >
        <div>
          <div
            style={
              Object {
                "textTransform": "capitalize",
              }
            }
          >
            domingo 04
          </div>
          <div
            style={
              Object {
                "color": "#aaaaaa",
                "textTransform": "capitalize",
              }
            }
          >
            muy nuboso
          </div>
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
        }
      }
    >
      <div>
        <div
          style={
            Object {
              "color": "#000",
            }
          }
        >
          27째
           
          <span
            style={
              Object {
                "color": "#aaaaaa",
                "marginLeft": "8px",
              }
            }
          >
            16째
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
